assets/js/collaborative-editor/components/inspector/JobInspector.tsx

```diff
@@ -345,6 +345,11 @@ export function JobInspector({ job, renderFooter }: JobInspectorProps) {
                   "keychain_credential_id",
                   resolved.keychain_credential_id
                 );
+                // Manually trigger updateJob to persist credentials to Y.Doc
+                updateJob(job.id, {
+                  project_credential_id: resolved.project_credential_id,
+                  keychain_credential_id: resolved.keychain_credential_id,
+                });
```

Comment:
Do we really have to do this? I thought that by setting the field value on the form would trigger the onChange at the top? Hmm, guess it makes sense. Just want to be sure.

Also there is a type warning here, Session.Job is missing project_credential_id, weird I'm not getting it for keychain_credential_id as well. But yeah we need to add the properties to the type.